### ISSUE TITLE
Add HSQL file persistence to default config to enable call caching on run mode and persistence of server mode

### ISSUE THREAD

**Issue by danbills** [broadinstitute/cromwell#3786]

**Comment by kshakir** [comment]
Something like this will be great for users who just want something simple. We may need to add warnings to the Cromwell docs depending on how this ticket is implemented.

The specific behavior this ticket aims to emulate / implement should be further refined with respect to HSQLDB. Plugging in `file:` will absolutely work for "hello world". But if one runs cromwell(s) the wrong way the db may become corrupted/deadlocked negating the ability to call-cache.

Many databases have minimal to no support for sharing an embedded instance between concurrent procs. SQLite has the most "support" afaik but a) would require _a lot_ of custom Cromwell code, and b) still has other issues such as in NFS environments. Depending on whomever this ticket is aimed at, if they're using an HPC environment like our methods users do we'd have to be careful not to store a multiprocess embedded DB on NFS.

Today with HSQLDB `mem:` cromwell uses a pair of ephemeral database connection pools. I'm not sure the behavior if both pools are pointed at the same HSQLDB `file:`, but I think it might work as the docs only warn of connecting from multi-process not multi-pool. The default config mentioned in this ticket may still consider using separate `file:` instances just in case.

All issues above have workarounds with varying degrees of difficulty and/or documentation warnings. For example one could clarify the documentation with "Cromwell only supports one instance connecting to the pair of default _file:_ databases at a time." Or: "Cromwell only supports call caching when running a workflow with the same name" because we did something like generate the db file based on the workflow name.

Another option, instead of having multiple processes access the same embedded DB, is to research spinning up a background daemon db process, which do support multiple connections.

Links to consider when defining acceptance criteria are below. 

Re: our existing/proposed HSQLDB usage
- Cromwell's `database.metadata` and `database.engine` when absent both [fall back to the root `database` stanza.](https://github.com/broadinstitute/cromwell/blob/088e12d97dd18f463e6a387a6ffb002d9725cbe4/services/src/main/scala/cromwell/services/ServicesStore.scala#L12)
- ["This allows each instance of a database object to use a clean, and different, in memory database."](https://github.com/broadinstitute/cromwell/blob/a8a605ed1f2f2d2de2db9b05c395a2c87ebfc295/database/sql/src/main/scala/cromwell/database/sql/SqlDatabase.scala#L17-L39)
- ["only one Java process at a time can make in-process connections to a given _file:_ database"](http://hsqldb.org/doc/guide/running-chapt.html#rgc_inprocess)
- ["Several different programs can connect to the server and retrieve or update information."](http://hsqldb.org/doc/2.0/guide/running-chapt.html#rgc_server_modes)

Re: SQLite
- ["You can't add a constraint to existing table in SQLite"](https://stackoverflow.com/a/15498225)
- ["use caution: this locking mechanism might not work correctly if the database file is kept on an NFS filesystem"](https://www.sqlite.org/faq.html#q5)
- We often change our table uniqueness using [Liquibase](https://github.com/broadinstitute/cromwell/pull/3553/files#diff-76feec217bb5aaed111d4c3c11ead546). HSQLDB stopped supporting [unique indexes years ago](http://www.hsqldb.org/doc/1.8/guide/ch09.html#create_index-section), so Cromwell only uses their cousin the [unique constraint](https://community.oracle.com/thread/1033157). Meanwhile SQLite does not allow adding/dropping a unique constraint without [copying the whole table](https://stackoverflow.com/a/42013422) but does support [adding/dropping a unique index](https://www.sqlite.org/lang_createindex.html).
- Additionally, SQLite will need a bunch of custom code for its [JDBC-lite implementation](https://stackoverflow.com/a/48643377).

**TL;DR Be sure of the use case before verifying that "hello world" passes, and consider labeling any implementation as "experimental"**
